site/client/figures/Figure.ts

```diff
@@ -0,0 +1,13 @@
+import { GlobalEntitySelection } from "../global-entity/GlobalEntitySelection"
+
+export abstract class Figure {
+    abstract get isLoaded(): boolean
```

Comment:
should `isLoaded`, `container`, `boundingRect` be abstract methods or implemented in this base class? seems like the implementations are the same in the derived classes

Reply:
They probably should, but it just feels inelegant and harder to understand.

Have to pass a subset of props to `super()` then store a prop on the abstract class (can't name it `props` or `container` because those are used in the children classes). You end up following a chain of names in order to get to the fact that `container` is simply passed from the `props`. 

I think repeating the implementation makes the code easier to understand. I've now turned `Figure` into an interface for that reason.

Also, I think @shaahmad can use this abstraction to change the prominent links to include the countries from the floating country select – then the `isLoaded` method would always return `true` for those I imagine, since we don't need to observe them on scroll and load on demand (observing lots of elements makes scrolling expensive).